test/support/dsl_controller.ex

```diff
@@ -0,0 +1,48 @@
+defmodule OpenApiSpexTest.DslController do
+  use Phoenix.Controller
+
+  import OpenApiSpex.OperationDsl
+
+  tags(["users"])
+
+  operation(:index,
+    summary: "User index",
+    parameters: [
+      query: [in: :query, type: :string, description: "Free-form query string", example: "jane"]
+    ]
+  )
+
+  def index(conn, _params) do
+    json(conn, %{
+      data: [
+        %{
+          id: "abc123",
+          name: "joe user",
+          email: "joe@gmail.com"
+        }
+      ]
+    })
+  end
+
+  operation(:show,
```

Comment:
This DSL looks good. 
I had a play around with some other syntaxes, eg

```elixir
@operation show: [
...
]
```

```elixir
operation :show do
...
end
```

But I think just dropping the parens makes it feel like a module level declaration, eg:

```elixir
  operation :show, [
    summary: "Show user",
    parameters: [
      id: [
        in: :path,
        description: "User ID",
        type: :integer,
        example: 1001
      ]
    ]
  ]
  def show(conn, %{id: id}) do
    json(conn, %{
      data: %{
        id: id,
        name: "joe user",
        email: "joe@gmail.com"
      }
    })
  end
```

And it will be easier find the `operation/2` macro in the docs vs a magic `@` attribute.

Reply:
I agree. I'll look into how to get OpenApiSpex to expose Elixir Formatter rules so that users can make one small change to the `import_deps: [ ... ]` list in their `.formatter.exs` file to allow optional parentheses for these macro calls.

Reply:
Elixir Formatter changes: 4cfdba26d5b77857697d701f081e90f5b40de8e7